From 0525bd82f6a9b5860ed3ea53520ac11ebfa09d5b Mon Sep 17 00:00:00 2001
Message-Id: <0525bd82f6a9b5860ed3ea53520ac11ebfa09d5b.1652325033.git.geliang.tang@suse.com>
From: Marco Elver <elver@google.com>
Date: Tue, 30 Nov 2021 12:44:30 +0100
Subject: [PATCH] objtool, kcsan: Add memory barrier instrumentation to
 whitelist
Git-commit: 0525bd82f6a9b5860ed3ea53520ac11ebfa09d5b
Patch-mainline: v5.17-rc1
References: bsn#258
Modified-by-SEL: No


Adds KCSAN's memory barrier instrumentation to objtool's uaccess
whitelist.

Signed-off-by: Marco Elver <elver@google.com>
Signed-off-by: Paul E. McKenney <paulmck@kernel.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/objtool/check.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/objtool/check.c b/tools/objtool/check.c
index 21735829b860..61dfb66b30b6 100644
--- a/tools/objtool/check.c
+++ b/tools/objtool/check.c
@@ -849,6 +849,10 @@ static const char *uaccess_safe_builtin[] = {
 	"__asan_report_store16_noabort",
 	/* KCSAN */
 	"__kcsan_check_access",
+	"__kcsan_mb",
+	"__kcsan_wmb",
+	"__kcsan_rmb",
+	"__kcsan_release",
 	"kcsan_found_watchpoint",
 	"kcsan_setup_watchpoint",
 	"kcsan_check_scoped_accesses",
-- 
2.34.1

